### PR TITLE
Feature/EPAY-59: Implement Elasticsearch

### PR DESCRIPTION
**Pull request by How-What**
Implemented Elasticsearch, same search functionality as before but now you can search 'billing name' with partial letters.

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -33,6 +37,8 @@ def create_app(config_name):
     bootstrap.init_app(app)
     db.init_app(app)
     login_manager.init_app(app)
+    app.elasticsearch = Elasticsearch([app.config['ELASTICSEARCH_URL']]) \
```
[the remark this answers is not in the document]
Comment (How-What) [reply app/__init__.py]: 'es.init_app(app)' doesn't work

File: app/search.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+                        },
+                        "suborder_number":{
+                            "type": 'keyword',
+                        },
+                        "order_number": {
+                            "type": 'keyword',
+                        },
+                        "order_type": {
+                            "type": 'keyword',
+                            "similarity":'boolean'
```
[the remark this answers is not in the document]
Comment (How-What) [reply app/search.py]: in this case nothing, it's been removed

File: app/search.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    date_range = {
+        'date_received_start': date_received_start,
+        'date_received_end': date_received_end,
+        'date_submitted_start': date_submitted_start,
+        'date_submitted_end': date_submitted_end
+    }
+
+    dsl_gen = DSLGenerator(query_fields=query_field, date_range=date_range)
+    dsl = dsl_gen.search() if any(query_field.values()) or any(date_range.values()) else dsl_gen.no_query()
+
```
Comment (gzhou1) [inline app/search.py]: Too many extra empty lines.

File: app/search.py [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+    date_range = {
+        'date_received_start': date_received_start,
+        'date_received_end': date_received_end,
+        'date_submitted_start': date_submitted_start,
+        'date_submitted_end': date_submitted_end
+    }
+
+    dsl_gen = DSLGenerator(query_fields=query_field, date_range=date_range)
+    dsl = dsl_gen.search() if any(query_field.values()) or any(date_range.values()) else dsl_gen.no_query()
```
Comment (gzhou1) [inline app/search.py]: Convert this to regular multiline if and else statement.

**Pull request merged**